jobs: TestStartableJob failed

[(jobs).TestStartableJob failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1940634&tab=buildLog) on [master@62f32ca9a726c84f9c9b516e97008bdf3253e1af](https://github.com/cockroachdb/cockroach/commits/62f32ca9a726c84f9c9b516e97008bdf3253e1af):

```
I200513 16:53:13.663702 22278 vendor/github.com/cockroachdb/pebble/open.go:270  [n?] [JOB 1] WAL created 000002
I200513 16:53:13.663870 22278 server/config.go:576  [n?] 1 storage engine initialized
I200513 16:53:13.664014 22278 server/config.go:579  [n?] Pebble cache size: 128 MiB
I200513 16:53:13.664122 22278 server/config.go:579  [n?] store 0: in-memory, size 0 B
I200513 16:53:13.668527 22278 server/server.go:1253  [n?] **** add additional nodes by specifying --join=127.0.0.1:38161
I200513 16:53:13.668601 22278 server/init.go:166  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200513 16:53:13.668651 22278 server/init.go:186  [n?] **** cluster 61f5085e-be3f-4268-8ac5-7433caa06247 has been created
I200513 16:53:13.668765 22278 gossip/gossip.go:405  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38161" > attrs:<> locality:<> ServerVersion:<major_val:20 minor_val:1 patch:0 unstable:4 > build_tag:"v20.1.0-beta.4-1545-g62f32ca9a7" started_at:1589388793668695526 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:39025" > 
I200513 16:53:13.668826 22340 server/server.go:1320  [n1] connecting to gossip network to verify cluster ID "61f5085e-be3f-4268-8ac5-7433caa06247"
I200513 16:53:13.688728 22340 server/server.go:1323  [n1] node connected via gossip
W200513 16:53:13.688814 22431 kv/kvserver/store.go:1652  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200513 16:53:13.691036 22278 server/node.go:418  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=16 KiB), ranges=32, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=166.00 pMax=13892.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200513 16:53:13.691226 22278 kv/kvserver/stores.go:236  [n1] read 0 node addresses from persistent storage
I200513 16:53:13.691329 22278 server/node.go:474  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200513 16:53:13.696271 22278 server/server.go:1415  [n1] starting https server at 127.0.0.1:43545 (use: 127.0.0.1:43545)
I200513 16:53:13.696315 22278 server/server.go:1420  [n1] starting postgres server at 127.0.0.1:39025 (use: 127.0.0.1:39025)
I200513 16:53:13.696343 22278 server/server.go:1422  [n1] starting grpc server at 127.0.0.1:38161
I200513 16:53:13.696358 22278 server/server.go:1423  [n1] advertising CockroachDB node at 127.0.0.1:38161
I200513 16:53:13.710642 22463 kv/kvserver/replica_rangefeed.go:608  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp is empty
I200513 16:53:13.712516 22371 kv/kvserver/queue.go:579  [n1,s1,r8/1:/Table/1{2-3}] rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I200513 16:53:13.712685 22473 sql/temporary_schema.go:472  [n1] running temporary object cleanup background job
I200513 16:53:13.713860 22473 sql/temporary_schema.go:507  [n1] found 0 temporary schemas
I200513 16:53:13.713892 22473 sql/temporary_schema.go:510  [n1] early exiting temporary schema cleaner as no temporary schemas were found
I200513 16:53:13.713904 22473 sql/temporary_schema.go:511  [n1] completed temporary object cleanup job
I200513 16:53:13.713918 22473 sql/temporary_schema.go:594  [n1] temporary object cleaner next scheduled to run at 2020-05-13 17:23:13.709681172 +0000 UTC
I200513 16:53:13.715841 22472 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:38161 Attrs: Locality: ServerVersion:20.1-4 BuildTag:v20.1.0-beta.4-1545-g62f32ca9a7 StartedAt:1589388793668695526 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:39025} ClusterID:61f5085e-be3f-4268-8ac5-7433caa06247 StartedAt:1589388793668695526 LastUp:1589388793668695526}
I200513 16:53:13.718852 22515 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200513 16:53:13.732243 22539 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:20.1-4 User:root}
I200513 16:53:13.747841 22396 vendor/github.com/cockroachdb/pebble/db.go:1300  [n1] [JOB 2] WAL created 000004
I200513 16:53:13.764319 22471 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200513 16:53:13.776310 22965 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:a1e2227b-1c8b-4f12-9fe3-0dfe9581fb1f User:root}
I200513 16:53:13.787031 22928 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200513 16:53:13.792311 22938 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200513 16:53:13.895331 22278 server/server_sql.go:640  [n1] done ensuring all necessary migrations have run
I200513 16:53:13.895382 22278 server/server.go:1733  [n1] serving sql connections
--- FAIL: TestStartableJob (0.46s)
=== RUN   TestStartableJob/Run_-_error_during_resume
I200513 16:53:13.994422 23031 jobs/registry.go:810  [n1] job 554822513118838785: stepping through state running with error <nil>
I200513 16:53:13.994773 23031 jobs/registry.go:810  [n1] job 554822513118838785: stepping through state reverting with error boom
I200513 16:53:14.001324 23031 jobs/registry.go:810  [n1] job 554822513118838785: stepping through state failed with error boom
E200513 16:53:14.009973 23031 jobs/registry.go:959  [n1] job 554822513118838785: adoption completed with error boom
I200513 16:53:14.011821 22494 jobs/registry.go:810  [n1] job 554822513415127041: stepping through state running with error <nil>
I200513 16:53:14.012154 22494 jobs/registry.go:810  [n1] job 554822513415127041: stepping through state succeeded with error <nil>
I200513 16:53:14.015308 23031 jobs/registry.go:965  [n1] job 554822513118838785: status failed after adoption finished
I200513 16:53:14.020521 22494 jobs/registry.go:965  [n1] job 554822513415127041: status succeeded after adoption finished
    --- FAIL: TestStartableJob/Run_-_error_during_resume (0.03s)
        jobs_test.go:2208: 
            	Error Trace:	jobs_test.go:2208
            	Error:      	Expect "<nil>" to match "boom"
            	Test:       	TestStartableJob/Run_-_error_during_resume

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestStartableJob PKG=./pkg/jobs TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestStartableJob.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -2092,10 +2092,11 @@ func TestStartableJob(t *testing.T) {
 	}
 	t.Run("Start called more than once", func(t *testing.T) {
 		sj := createStartableJob(t)
-		_, err := sj.Start(ctx)
+		errCh, err := sj.Start(ctx)
 		require.NoError(t, err)
 		_, err = sj.Start(ctx)
 		require.Regexp(t, `StartableJob \d+ cannot be started more than once`, err)
+		require.NoError(t, <-errCh)
 	})
 	t.Run("Start called with active txn", func(t *testing.T) {
 		txn := db.NewTxn(ctx, "test")

